### PR TITLE
chia-dev-tools: init at 1.1.4

### PR DESCRIPTION
**Pull request by bbjubjub2494**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/blockchains/chia-dev-tools/default.nix [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+  disabledTests = [
+    "test_spendbundles"
+  ];
+
+  meta = with lib; {
+    homepage = "https://www.chia.net/";
+    description = "A utility for developing in the Chia ecosystem: Chialisp functions, object inspection, RPC client and more";
+    license = with licenses; [asl20];
+    maintainers = teams.chia.members;
+    platforms = platforms.all;
```
Comment (SuperSandro2000) [inline pkgs/applications/blockchains/chia-dev-tools/default.nix:61]:
```suggestion
```
buildPythonApplication defaults to linux+darwin and only supports that

File: pkgs/applications/blockchains/chia-dev-tools/default.nix [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+  postCheck = "unset HOME";
+
+  disabledTests = [
+    "test_spendbundles"
+  ];
+
+  meta = with lib; {
+    homepage = "https://www.chia.net/";
+    description = "A utility for developing in the Chia ecosystem: Chialisp functions, object inspection, RPC client and more";
+    license = with licenses; [asl20];
```
Comment (SuperSandro2000) [inline pkgs/applications/blockchains/chia-dev-tools/default.nix:59]:
```suggestion
    description = "Utility for developing in the Chia ecosystem: Chialisp functions, object inspection, RPC client and more";
    license = with licenses; [ asl20 ];
```

File: pkgs/applications/blockchains/chia-dev-tools/default.nix [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    pytest-asyncio
+    pytimeparse
+  ];
+
+  checkInputs = with python3Packages; [
+    pytestCheckHook
+  ];
+
+  preCheck = "export HOME=$(mktemp -d)";
+  postCheck = "unset HOME";
```
Comment (SuperSandro2000) [inline pkgs/applications/blockchains/chia-dev-tools/default.nix:50]:
```suggestion
  preCheck = ''
    export HOME=$(mktemp -d)
  '';
```
we don't need to unset HOME

File: pkgs/applications/blockchains/chia-dev-tools/default.nix [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+  nativeBuildInputs = [
+    python3Packages.setuptools-scm
+  ];
+
+  # give a hint to setuptools-scm on package version
+  SETUPTOOLS_SCM_PRETEND_VERSION = "v${version}";
+
+  propagatedBuildInputs = with python3Packages; [
+    (toPythonModule chia)
+    pytest
```
Comment (SuperSandro2000) [inline pkgs/applications/blockchains/chia-dev-tools/default.nix:40]:
```suggestion
```
pytest is included in pytestCheckHook already and should be convention never be propagated

File: pkgs/applications/blockchains/chia-dev-tools/default.nix [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+    pytest-asyncio
```
Comment (SuperSandro2000) [inline pkgs/applications/blockchains/chia-dev-tools/default.nix:41]: You are sure this is required at runtime?

File: pkgs/applications/blockchains/chia-dev-tools/default.nix [hunk 6]
```diff
@@ -0,0 +1,63 @@
+{
+  lib,
+  fetchFromGitHub,
+  substituteAll,
+  python3Packages,
+  chia,
+}:
```
Comment (SuperSandro2000) [inline pkgs/applications/blockchains/chia-dev-tools/default.nix:7]:
```suggestion
{ lib
, fetchFromGitHub
, substituteAll
, python3Packages
, chia
}:

```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by lourkeur** [comment]
Note to self: use `nixpkgs-fmt` instead of `alejandra` in this project.

**Pull request merged**